hedera-node/hedera-app/src/main/java/com/hedera/node/app/state/merkle/StringLeaf.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright (C) 2022 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hedera.node.app.state.merkle;
+
+import com.swirlds.common.io.streams.SerializableDataInputStream;
+import com.swirlds.common.io.streams.SerializableDataOutputStream;
+import com.swirlds.common.merkle.MerkleLeaf;
+import com.swirlds.common.merkle.impl.PartialMerkleLeaf;
+import com.swirlds.common.utility.CommonUtils;
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+import java.io.IOException;
+import java.util.Objects;
+
+/**
+ * A {@link MerkleLeaf} containing a single String value. This is an immutable leaf -- once set, the
+ * leaf value can never change. The maximum size for the leaf value is 128 bytes.
+ */
+final class StringLeaf extends PartialMerkleLeaf implements MerkleLeaf {
+    // For serialization
+    private static final long CLASS_ID = 99992323882382L;
+    private static final int VERSION_0 = 0;
+    private static final int CURRENT_VERSION = VERSION_0;
+
+    /** The maximum length we permit for the string's value. This is in bytes, not in chars. */
+    private static final int MAX_LENGTH = 128;
+
+    /**
+     * The value of the leaf. Will never be null, and will never have more than {@link #MAX_LENGTH}
+     * bytes. Normally, this would be final, but the deserialization system of the hashgraph
+     * platform does not permit it to be final.
+     */
+    private String value = "";
+
+    /**
+     * @deprecated Used by the deserialization system only
+     */
+    @Deprecated(since = "1.0")
+    public StringLeaf() {}
+
+    /**
+     * Create a new instance with the given value for the leaf.
+     *
+     * @param value The value cannot be null.
+     */
+    public StringLeaf(@NonNull final String value) {
+        Objects.requireNonNull(value);
+
+        // Scrub the data by converting to normalized bytes, and check that we didn't get more bytes
+        // than we should.
+        final byte[] data = CommonUtils.getNormalisedStringBytes(value);
+        if (data.length > MAX_LENGTH) {
+            throw new IllegalArgumentException(
+                    "The maximum *normalized* string length allowed is " + MAX_LENGTH);
```

Comment:
I think, it would help, if you include the actual length in the error message.

Reply:
This class has now been removed.